### PR TITLE
Optimization for memory allocation

### PR DESCRIPTION
**Pull request by makeecat**
Tested with BinB0 and Binworld with arpl_autonomy if the received odom are correctly used.

### CODE REVIEW

File: src/transform.cpp [hunk 1]
```diff
@@ -3,67 +3,65 @@
...
-  //   }
-  //   else {
-  //     PRINT_INFO("Failed to get init T_BtoW from vicon topic %s, use default T_BtoW by setting init_world_with_vicon false\n", viconOdomWTopic.c_str());
-  //     exit(1);
-  //   }
-  // } 
+    nh->getParam("init_world_with_vicon", init_world_with_vicon);
+
+    // TODO: Check this part with Vicon later  
+    bool init_world_with_vicon = false;
```
Comment (xjh19971) [inline src/transform.cpp]: This is always overwritten by "false"?

File: src/transform.cpp [hunk 2]
```diff
@@ -73,83 +71,69 @@ void Transform_calculator::setupTransformationMatrix(){
...
+    // T_ItoB0 = T_MtoB0 * T_ItoM;
+    //T_ItoW = T_MtoW * T_ItoM;
+
     T_ItoM = T_init_tf_inv * T_ItoM;
-    // The TF that is required for flight test
-    Eigen::Matrix4d T_BtoB0 = (T_ItoB * T_ItoM * T_BtoI);
-    
-    // Transform the body pose in World frame
-    Eigen::Matrix4d T_BtoW = Eigen::Matrix4d::Identity();
+    T_BtoB0 = T_ItoB * T_ItoM * T_BtoI;
```
Comment (xjh19971) [inline src/transform.cpp:119]: No B0 in the right hand side?

File: src/transform.cpp (hunk 1 above)
Comment (makeecat) [reply src/transform.cpp]: That has been fixed.

File: src/transform.cpp (hunk 2 above)
Comment (makeecat) [reply src/transform.cpp:119]: > No B0 in the right hand side?
Because T_ItoB = T_I0toB0

**Review by xjh19971 — APPROVED** [review]

**Pull request merged**